File: src/MeloTalentAI.Web/src/api/jobAnalysis.ts
import { api } from "./client";

export interface AnalyzeJobRequest {
  resumeId?: string;
  jobTitle: string;
  companyName: string;
  jobDescription: string;
  sourceUrl?: string;
}

export interface AnalyzeJobResponse {
  jobApplicationId: string;
  analysisId: string;
  jobTitle: string;
  companyName: string;
  matchScore: number;
  matchingSkills: string[];
  missingSkills: string[];
  recommendations: string;
  generatedCoverLetter?: string;
  interviewQuestions: string[];
}

export async function analyzeJob(
  request: AnalyzeJobRequest,
): Promise<AnalyzeJobResponse> {
  const response = await api.post<AnalyzeJobResponse>(
    "/api/job-analysis",
    request,
  );

  return response.data;
}

File: src/MeloTalentAI.Api/Domain/Entities/JobAnalysis.cs
namespace MeloTalentAI.Api.Domain.Entities;

public class JobAnalysis
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid JobApplicationId { get; set; }

    public int MatchScore { get; set; }

    public List<string> MatchingSkills { get; set; } = [];

    public List<string> MissingSkills { get; set; } = [];

    public string Recommendations { get; set; } = string.Empty;

    public string? GeneratedCoverLetter { get; set; }

    public List<string> InterviewQuestions { get; set; } = [];

    public DateTime AnalyzedAtUtc { get; set; } = DateTime.UtcNow;

    public JobApplication JobApplication { get; set; } = null!;
}
